### PR TITLE
feat!: upgrade to protopedia-api-v2-client v3.0.0 and improve module exports

### PR DESCRIPTION
**Pull request by F88**
## Summary

Upgrades the library to support `protopedia-api-v2-client` v3.0.0, handling breaking changes where 9 API response fields changed from required to optional.

### API Client Version

- Updated `peerDependencies` from `^2.0.0` to `^3.0.0`

### Optional Fields (v3.0.0)

The following fields in `ResultOfListPrototypesApiResponse` are now optional:
- `teamNm`, `users`, `freeComment`, `releaseDate`, `thanksFlg`, `uuid`, `revision`, `releaseFlg`, `licenseType`

### Module Exports (Standalone Usability)

- **BREAKING**: Type definitions (`StatusCode`, `ReleaseFlagCode`, `LicenseTypeCode`, `ThanksFlagCode`) are no longer exported from `@f88/promidas/utils`
  - Import from `@f88/promidas/types` instead
  - Utils module now focuses on converter functions and time utilities

### Type Definitions

- Updated `NormalizedPrototype` type to use `Type?: undefined | Type` pattern for strict TypeScript `exactOptionalPropertyTypes` compatibility
- All optional properties now explicitly handle `undefined` values

### Data Normalization

Refactored `normalizePrototype()` function to apply appropriate defaults for optional fields:
- `teamNm`: `''` (empty string)
- `users`: `[]` (empty array)  
- `freeComment`: `''` (empty string)
- `releaseDate`: `undefined` when missing
- `thanksFlg`: `0`
- `revision`: `0`
- `releaseFlg`: `2` (Released status)
- `licenseType`: `1`
- `uuid`: preserved as-is (may be `undefined`)

### Module Exports (New)

Enhanced standalone usability of `lib/*` modules:
- Added code type exports to `@f88/promidas/types`
- Added `NormalizedPrototype` re-exports to `@f88/promidas/store` and `@f88/promidas/repository`
- Added `Logger`/`LogLevel` re-exports to `@f88/promidas/repository`
- Removed type re-exports from `@f88/promidas/utils` for clearer responsibility separation

Each module can now be used independently without requiring additional imports from other modules.

### Testing

- Added 43 new test cases for v3.0.0 optional field handling
- Total test count: 485 → 501 tests
- Coverage includes edge cases, extreme values, Unicode/emoji, boundary conditions
- All tests passing ✅

### Documentation

- Created `CHANGELOG.md` with detailed v3.0.0 upgrade information
- Updated `README.md` with API client version compatibility notes and subpath exports guide
- Enhanced `lib/fetcher/docs/USAGE.md` with v3.0.0 migration guide
- Improved TSDoc comments in `lib/fetcher/index.ts`, `lib/types/normalized-prototype.ts`, and `lib/fetcher/utils/normalizers.ts`
- Fixed all module READMEs to match actual implementation
  - `lib/utils`: Updated to reflect actual label converter functions
  - `lib/types`: Added code types documentation
  - `lib/logger`: Fixed function name typo
  - `lib/fetcher`: Fixed field name in examples
  - `lib/store`: Fixed class usage and method names
  - `lib/repository`: Complete rewrite to match actual API
- Added standalone import examples to all module READMEs for better discoverability

## Compatibility

- ✅ Fully backward compatible - no code changes required for consumers (except utils type imports)
- ✅ Automatic handling of optional fields with sensible defaults
- ✅ Type-safe with `exactOptionalPropertyTypes: true`
- ✅ All existing tests pass
- ✅ Build successful

## Testing Results

```
Test Files  24 passed (24)
Tests  501 passed (501)
```

Type checking: ✅ No errors
Build: ✅ Successful

## Related

- Upstream release: https://github.com/F88/protopedia-api-v2-client.js/releases/tag/v3.0.0

### CODE REVIEW

File: lib/types/normalized-prototype.ts [hunk 1]
```diff
@@ -1,436 +1,110 @@
...
-   * ]
-   * ```
-   */
-  materials?: string[];
+  /* Others */
+  uuid?: undefined | string;
+  nid?: undefined | string;
+  revision?: undefined | number;
+  licenseType?: undefined | number;
+  thanksFlg?: undefined | number;
```
[the remark this answers is not in the document]
Comment (F88) [reply lib/types/normalized-prototype.ts]: NO
